code/game/gamemodes/roleset/roleset.dm

```diff
@@ -43,32 +48,72 @@
+			return list()
+
 	var/list/candidates = list()
+	var/any_candidates = FALSE
+	var/mob/living/L
+
 	for(var/datum/mind/candidate in SSticker.minds)
-		if(!candidate.current)
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] has no mob"))
+		L = candidate.current
+		if(request_log.Find(candidate.key))
+			var/last_request = request_log[candidate.key]
+			if((world.time - last_request) < request_timeout)
+				if(report)
+					to_chat(report, SPAN_NOTICE("Failure: [candidate] was already asked too recently"))
+				continue
+		if(!L)
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] has no mob"))
 			continue
 		if(!temp.can_become_antag(candidate, report))
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] can't become this antag"))
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] can't become this antag"))
 			continue
 		if(!antagonist_suitable(candidate, temp))
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] is not antagonist suitable"))
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] is not antagonist suitable"))
 			continue
-		if(!(temp.bantype in candidate.current.client.prefs.be_special_role))
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] has special role [temp.bantype] disabled"))
+		if(!(temp.bantype in L.client.prefs.be_special_role))
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] has special role [temp.bantype] disabled"))
 			continue
 		if(GLOB.storyteller && GLOB.storyteller.one_role_per_player && candidate.antagonist.len)
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] is already a [candidate.antagonist[1]] and can't be two antags"))
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] is already a [candidate.antagonist[1]] and can't be two antags"))
 			continue
 		if(player_is_antag_id(candidate,antag))
-			if (report) to_chat(report, SPAN_NOTICE("Failure: [candidate] is already a [antag]"))
+			if(report)
+				to_chat(report, SPAN_NOTICE("Failure: [candidate] is already a [antag]"))
 			continue
+
 		candidates.Add(candidate)
+		any_candidates = TRUE
+
+		//Activity test
+		if(act_test)
+			spawn()
+				if(!request_log.Find(candidate.key))
+					request_log.Add(candidate.key)
+				request_log[candidate.key] = world.time //Record this request so we don't spam them repeatedly
+				usr = L
+				usr << 'sound/effects/magic/blind.ogg' //Play this sound to a player whenever when he's chosen to decide.
+				to_chat(usr, "<font color='purple'>You are a candidate for [temp.role_text].\
+								<br>If you wish to opt-out of the candidate pool, click <a HREF=?src=\ref[src];opt_out=\ref[candidate.key]>here</a> within 60 seconds.\
+								<br>Otherwise, do nothing and you will be added to the list of candidates.</font>")
+				sleep(60 SECONDS)
```

Comment:
Could this be converted to use a callback, instead of sleeping for 60 seconds?

Reply:
It's doable. What's the benefit of using callback over sleep?